src/Symfony/Bundle/FrameworkBundle/CacheWarmer/SerializerCacheWarmer.php

```diff
@@ -41,8 +41,8 @@ public function __construct(array $loaders, string $phpArrayFile)
 
     protected function doWarmUp(string $cacheDir, ArrayAdapter $arrayAdapter): bool
     {
-        if (!class_exists(CacheClassMetadataFactory::class) || !method_exists(XmlFileLoader::class, 'getMappedClasses') || !method_exists(YamlFileLoader::class, 'getMappedClasses')) {
-            return false;
+        if (!$this->loaders) {
```

Comment:
In theory, you could've created an instance of this cache warmer with an empty loaders array without the serializer being installed and the feature detection would've prevented it from blowing up.

Since this warmer doesn't do anything with an empty loaders array anyway, I use this check for an early exit instead.